### ISSUE TITLE
[0.9.0rc32] default retention policy defaults to 168h0m0s instead of 0? 

### ISSUE THREAD

**Issue by jonaz** [influxdb/influxdb#2817]
In r31 the retention defaulted to duration 0 when i created new databases. 
After generating config which looks like this newly created databases get retention duration 168h0m0s. 168h0m0s seems to be defined in the hinted-handoff, but should that affect retention? 

reproduce:
rm -rf /root/.influxdb/* (clear the data dirs to start fresh)
curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE testdb"
curl -G http://localhost:8086/query --data-urlencode "q=SHOW RETENTION POLICIES testdb"

this is my config: 
```
[meta]
  dir = "/root/.influxdb/meta"
  hostname = "localhost"
  bind-address = ":8088"
  retention-autocreate = true
  election-timeout = "1s"
  heartbeat-timeout = "1s"
  leader-lease-timeout = "500ms"
  commit-timeout = "50ms"

[data]
  dir = "/root/.influxdb/data"
  retention-auto-create = true
  retention-check-enabled = true
  retention-check-period = "10m0s"
  retention-create-period = "45m0s"

[cluster]
  shard-writer-timeout = "5s"

[retention]
  enabled = true
  check-interval = "10m0s"

[admin]
  enabled = true
  bind-address = ":8083"

[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = false
  log-enabled = true
  write-tracing = false
  pprof-enabled = false

[collectd]
  enabled = false
  bind-address = ""
  database = ""
  typesdb = ""

[opentsdb]
  enabled = false
  bind-address = ""
  database = ""
  retention-policy = ""

[udp]
  enabled = false
  bind-address = ""
  database = ""
  batch-size = 0
  batch-timeout = "0"

[monitoring]
  enabled = false
  write-interval = "1m0s"

[continuous_queries]
  enabled = true
  recompute-previous-n = 2
  recompute-no-older-than = "10m0s"
  compute-runs-per-interval = 10
  compute-no-more-than = "2m0s"

[hinted-handoff]
  enabled = true
  dir = "/root/.influxdb/hh"
  max-size = 1073741824
  max-age = "168h0m0s"
  retry-rate-limit = 0
  retry-interval = "1s"

```

**Comment by otoolep** [comment]
I'll take a look at this. In the meantime you can set the retention period for the automatically-created policy, to what it was before, using the CLI and the value `INF`.

**Comment by kiwicmc** [comment]
I tried this via the web interface and got a panic

Issued this: alter retention policy default on lun_metrics_test duration inf

GOMAXPROCS set to 4
2015/06/08 23:43:22 [INFO] raft: Node at 127.0.0.1:8088 [Follower] entering Follower state
2015/06/08 23:43:22 read local node id: 1
[httpd] 2015/06/08 23:43:22 listening on HTTP: [::]:8086
2015/06/08 23:43:23 [WARN] raft: Heartbeat timeout reached, starting election
2015/06/08 23:43:23 [INFO] raft: Node at 127.0.0.1:8088 [Candidate] entering Candidate state
2015/06/08 23:43:23 [DEBUG] raft: Votes needed: 1
2015/06/08 23:43:23 [DEBUG] raft: Vote granted. Tally: 1
2015/06/08 23:43:23 [INFO] raft: Election won. Tally: 1
2015/06/08 23:43:23 [INFO] raft: Node at 127.0.0.1:8088 [Leader] entering Leader state
2015/06/08 23:43:23 [INFO] raft: Disabling EnableSingleNode (bootstrap)
2015/06/08 23:43:23 [DEBUG] raft: Node 127.0.0.1:8088 updated peer set (2): [127.0.0.1:8088]
2015/06/08 23:43:23 [DEBUG] raft: Node 127.0.0.1:8088 updated peer set (2): [127.0.0.1:8088]
[http] 2015/06/08 23:43:35 203.109.194.21 - - [08/Jun/2015:23:43:35 +0000] GET /query?u=root&p=root&q=SHOW%20DATABASES HTTP/1.1 200 76 http://ec2-54-212-253-146.us-west-2.compute.amazonaws.com:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 2df3a9f4-0e38-11e5-8001-000000000000 2.032329ms
[http] 2015/06/08 23:43:44 203.109.194.21 - - [08/Jun/2015:23:43:44 +0000] GET /query?u=root&p=root&q=CREATE%20DATABASE%20lun_metrics_test HTTP/1.1 200 40 http://ec2-54-212-253-146.us-west-2.compute.amazonaws.com:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 335dbf82-0e38-11e5-8002-000000000000 5.88663ms
[http] 2015/06/08 23:43:45 203.109.194.21 - - [08/Jun/2015:23:43:45 +0000] GET /query?u=root&p=root&q=SHOW%20DATABASES HTTP/1.1 200 103 http://ec2-54-212-253-146.us-west-2.compute.amazonaws.com:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 3393e110-0e38-11e5-8003-000000000000 1.172061ms
[http] 2015/06/08 23:44:01 203.109.194.21 - - [08/Jun/2015:23:44:01 +0000] GET /query?u=root&p=root&q=show%20retention%20on%20lun_metrics_test&db=lun_metrics_test HTTP/1.1 400 99 http://ec2-54-212-253-146.us-west-2.compute.amazonaws.com:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 3d718313-0e38-11e5-8004-000000000000 1.337739ms
[http] 2015/06/08 23:44:15 203.109.194.21 - - [08/Jun/2015:23:44:15 +0000] GET /query?u=root&p=root&q=show%20retention%20policies%20lun_metrics_test&db=lun_metrics_test HTTP/1.1 200 120 http://ec2-54-212-253-146.us-west-2.compute.amazonaws.com:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36 45ef6bc7-0e38-11e5-8005-000000000000 1.628888ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x51d702]

goroutine 50 [running]:
github.com/influxdb/influxdb/meta.(*Store).UpdateRetentionPolicy(0xc20802c000, 0xc2080ca460, 0x10, 0xa36c50, 0x7, 0xc208134fc0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:719 +0xd2
github.com/influxdb/influxdb/meta.(*StatementExecutor).executeAlterRetentionPolicyStatement(0xc2080cd9e0, 0xc2080b7a40, 0xc2080b7a40)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/statement_executor.go:178 +0x10f
github.com/influxdb/influxdb/meta.(*StatementExecutor).ExecuteStatement(0xc2080cd9e0, 0x7f74bdec1a80, 0xc2080b7a40, 0xc20811e0fc)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/statement_executor.go:62 +0x4f5
github.com/influxdb/influxdb/tsdb.func·004()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:188 +0x674
created by github.com/influxdb/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:211 +0x22b

goroutine 1 [chan receive (nil chan), 2 minutes]:
main.(*Main).Run(0xc20816ff60, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:60 +0x3d6
main.main()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:24 +0x1d0

goroutine 6 [syscall, 2 minutes]:
os/signal.loop()
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, 2 minutes, locked to thread]:
runtime.goexit()
	/root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [IO wait, 2 minutes]:
net.(*pollDesc).Wait(0xc2080103e0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080103e0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010380, 0x0, 0x7f74bdeb9de0, 0xc20802a310)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20802e470, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20802e470, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080b9950, 0x7f74bdebbf40, 0xc20802e470, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:45 +0x57
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:210 +0x76b

goroutine 9 [chan receive, 2 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802e478, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc2080b9980, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:1539 +0x64
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc208010460)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:346 +0x50
created by github.com/hashicorp/raft.NewNetworkTransport
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:138 +0x2c4

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:807 +0xc7c
github.com/hashicorp/raft.(*Raft).runLeader(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:779 +0x74e
github.com/hashicorp/raft.(*Raft).run(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:576 +0xc9
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:241 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 11 [select, 1 minutes]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:490 +0xf74
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runFSM)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:242 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 12 [select, 2 minutes]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:1588 +0x466
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runSnapshots)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:243 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 19 [chan receive, 2 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802e480, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc20802c000)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:416 +0x8f
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:175 +0x196

goroutine 20 [select, 2 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc208098090, 0xc2080301e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:271 +0x149
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:216 +0x8dc

goroutine 21 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc20803f0e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:92 +0x2e7
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:61 +0xef

goroutine 22 [select, 2 minutes]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc20803f0e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:110 +0x2a4
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:62 +0x109

goroutine 23 [chan receive, 2 minutes]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802e488, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc208010230)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:79 +0x102
created by github.com/influxdb/influxdb/cluster.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:56 +0x67

goroutine 24 [IO wait, 1 minutes]:
net.(*pollDesc).Wait(0xc2081200d0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2081200d0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208120070, 0x0, 0x7f74bdeb9de0, 0xc2080cc058)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20810e018, 0xc20813ed90, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20810e018, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc208108180, 0x7f74bdebbf40, 0xc20810e018, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f74bdebbf40, 0xc20810e018, 0x7f74bdebd6e0, 0xc20802aa00, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc2080b97a0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:71 +0x147
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:39 +0xd9

goroutine 25 [select]:
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc2080102a0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:146 +0x1fc
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:87 +0x1ed

goroutine 26 [IO wait, 1 minutes]:
net.(*pollDesc).Wait(0xc208120140, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208120140, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc2081200e0, 0x0, 0x7f74bdeb9de0, 0xc2080cc0f0)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20810e020, 0xc2080eddc8, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20810e020, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc208031200, 0x7f74bdebbf40, 0xc20810e020, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f74bdebbf40, 0xc20810e020, 0x7f74bdebd640, 0xc208010310, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc20808f7c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:83 +0x79
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:51 +0x283

goroutine 27 [select, 2 minutes]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc208030780)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:64 +0x433
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x67

goroutine 28 [select, 2 minutes]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc208030780)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:98 +0x821
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:47 +0x81

goroutine 29 [select, 2 minutes]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc20808fa00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:110 +0x2d9
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:94 +0x87a

goroutine 30 [IO wait, 1 minutes]:
net.(*pollDesc).Wait(0xc2081201b0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2081201b0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208120150, 0xc208178000, 0x1000, 0x1000, 0x0, 0x7f74bdeb9de0, 0xc20814c088)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20810e038, 0xc208178000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/net.go:121 +0xdc
net/http.(*liveSwitchReader).Read(0xc208174048, 0xc208178000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:214 +0xab
io.(*LimitedReader).Read(0xc20810b0a0, 0xc208178000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/io/io.go:408 +0xce
bufio.(*Reader).fill(0xc208108720)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc208108720, 0xc2080e9b0a, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:295 +0x257
bufio.(*Reader).ReadLine(0xc208108720, 0x0, 0x0, 0x0, 0xc207fee500, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/bufio/bufio.go:324 +0x62
net/textproto.(*Reader).readLineSlice(0xc20811a270, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/textproto/reader.go:55 +0x9e
net/textproto.(*Reader).ReadLine(0xc20811a270, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/textproto/reader.go:36 +0x4f
net/http.ReadRequest(0xc208108720, 0xc20814a0d0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/request.go:598 +0xcb
net/http.(*conn).readRequest(0xc208174000, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:586 +0x26f
net/http.(*conn).serve(0xc208174000)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1162 +0x69e
created by net/http.(*Server).Serve
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1751 +0x35e

goroutine 32 [chan receive]:
github.com/influxdb/influxdb/services/httpd.(*Handler).serveQuery(0xc208010310, 0x7f74bdec0ef8, 0xc208134f60, 0xc208034ea0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:241 +0x7c4
github.com/influxdb/influxdb/services/httpd.*Handler.(github.com/influxdb/influxdb/services/httpd.serveQuery)·fm(0x7f74bdec0ef8, 0xc208134f60, 0xc208034ea0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:90 +0x4f
github.com/influxdb/influxdb/services/httpd.func·001(0x7f74bdec0ef8, 0xc208134f60, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:589 +0x78
net/http.HandlerFunc.ServeHTTP(0xc20801f580, 0x7f74bdec0ef8, 0xc208134f60, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·002(0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:647 +0x288
net/http.HandlerFunc.ServeHTTP(0xc2080cda80, 0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·003(0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:656 +0xb0
net/http.HandlerFunc.ServeHTTP(0xc20801f5e0, 0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·004(0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:689 +0x31a
net/http.HandlerFunc.ServeHTTP(0xc2080cdac0, 0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·005(0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:699 +0x179
net/http.HandlerFunc.ServeHTTP(0xc2080cdae0, 0x7f74bdec0e70, 0xc208134ec0, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·006(0x7f74bdec0e70, 0xc208134ea0, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:707 +0x123
net/http.HandlerFunc.ServeHTTP(0xc20801f600, 0x7f74bdec0e70, 0xc208134ea0, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·007(0x7f74bdebdb90, 0xc20815c000, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:717 +0x12c
net/http.HandlerFunc.ServeHTTP(0xc20801f620, 0x7f74bdebdb90, 0xc20815c000, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP(0xc20802e3c0, 0x7f74bdebdb90, 0xc20815c000, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/bmizerany/pat/mux.go:109 +0x21c
github.com/influxdb/influxdb/services/httpd.(*Handler).ServeHTTP(0xc208010310, 0x7f74bdebdb90, 0xc20815c000, 0xc208034ea0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:166 +0x256
net/http.serverHandler.ServeHTTP(0xc208031200, 0x7f74bdebdb90, 0xc20815c000, 0xc208034ea0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc20815c0a0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1751 +0x35e
root@ip-10-207-9-44:/var/log/influxdb#

**Comment by otoolep** [comment]
Fixed by PR #2838 2838

**Comment by otoolep** [comment]
Thanks for the bug report @jonaz -- all fixed.
